lib/internal/loader/Loader.js

```diff
@@ -24,112 +25,75 @@ function getURLStringForCwd() {
   }
 }
 
-/* A Loader instance is used as the main entry point for loading ES modules.
- * Currently, this is a singleton -- there is only one used for loading
- * the main module and everything in its dependency graph. */
-class Loader {
-  constructor(base = getURLStringForCwd()) {
-    if (typeof base !== 'string') {
-      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'base', 'string');
-    }
+// A closure is used here because we really really really don't want users
+// touching the hooks without using `hook()`.
+function Loader(base = getURLStringForCwd()) {
```

Comment:
Why not a class?

Reply:
the comment explains

Reply:
It can still be a class with the `this.hook =` in the constructor